crates/circuit/src/dag_circuit.rs

```diff
@@ -633,19 +721,19 @@ impl DAGCircuit {
         }
 
         let binding = dict_state.get_item("qubits")?.unwrap();
-        let qubits_raw = binding.downcast::<PyList>().unwrap();
-        for bit in qubits_raw.iter() {
-            self.qubits.add(py, &bit, false)?;
+        let qubits_raw = binding.extract::<Vec<ShareableQubit>>().unwrap();
```

Comment:
Isn't this throwing away a potential python space error now. I'd have expected:

```suggestion
        let qubits_raw = binding.extract::<Vec<ShareableQubit>>()?;
```

I think it was `unwrap()`  before to do the fast path with `downcast` and skip the error handling.

Reply:
Done in 5347aedca.